lib/checkcondition.cpp

```diff
@@ -1464,6 +1464,8 @@ void CheckCondition::alwaysTrueFalse()
     const SymbolDatabase *symbolDatabase = mTokenizer->getSymbolDatabase();
     for (const Scope * scope : symbolDatabase->functionScopes) {
         for (const Token* tok = scope->bodyStart->next(); tok != scope->bodyEnd; tok = tok->next()) {
+            if (tok->scope()->isClassOrStructOrUnion())
```

Comment:
Class or struct scopes shouldn't be added to `functionScopes`.

Reply:
Do you mean the tokens should be removed from the scope?

Reply:
> The outer scope is a function scope, but it has a local type definition.

I see. I didnt think about that.